src/iexamine.cpp

```diff
@@ -1897,15 +1897,17 @@ void iexamine::plant_seed( player &p, const tripoint &examp, const itype_id &see
     } else {
         used_seed = p.use_amount( seed_id, 1 );
     }
-    used_seed.front().set_age( 0_turns );
-    g->m.add_item_or_charges( examp, used_seed.front() );
-    if( g->m.has_flag_furn( "PLANTABLE", examp ) ) {
-        g->m.furn_set( examp, furn_str_id( g->m.furn( examp )->plant->transform ) );
-    } else {
-        g->m.set( examp, t_dirt, f_plant_seed );
+    if( !used_seed.empty() ) {
+        used_seed.front().set_age( 0_turns );
+        g->m.add_item_or_charges( examp, used_seed.front() );
+        if( g->m.has_flag_furn( "PLANTABLE", examp ) ) {
+            g->m.furn_set( examp, furn_str_id( g->m.furn( examp )->plant->transform ) );
+        } else {
+            g->m.set( examp, t_dirt, f_plant_seed );
+        }
+        p.moves -= to_moves<int>( 30_seconds );
```

Comment:
This is roughly 30 turns, during which the player cannot do anything. This let the player get attacked and not do any for 30 turns. I see that this isn't your doing, but perhaps it should be adjusted. That may be out of scope of this PR though.

Reply:
Yeah that was pre-existing code, I had to make a new activity for churning earth, as I couldnt just subtract 3 minutes worth of moves, 30 seconds is probably too much, Id say a maximum of a few seconds for a task, otherwise it needs an activity.

Out of scope for this PR but yes, there are probably a number of things that just subtract moves that should be activities.